### ISSUE TITLE
"DSA cn=toucan has started on Internet=127.0.0.1+17003" logged twice

### ISSUE THREAD

**Issue by JonathanWilbur** [Wildboar-Software/isode#1]
I don't see why this is happening. It only appears once in the `main()` function, and both the `main()` function and `compat/isoaddrs.c` haven't been changed. `paddr2str` is an alias for `_paddr2str`, which is in `compat/isoaddrs.c`. I spent 30 minutes looking at it with no luck. I'm thinking the right approach now is just to bisect it.

**Comment by JonathanWilbur** [comment]
I recently discovered that this happens because there are actually two NSAPs parsed from what is really a single address. I _also_ discovered that this seems to happen when you use a hostname instead of an IP address.

**Comment by JonathanWilbur** [comment]
I discovered that this was not caused by ISODE at all, but rather, by the `gethostbyname` system call. It was returning duplicate IP addresses in the `h_addr_list` field. I believe it was because `/etc/hosts` had two entries for `localhost`: an IPv4 and IPv6. Even though `gethostbyname` returned two IP addresses, I believe it translated the IPv6 address to the equivalent IPv4 and returned the result. When I removed the IPv6 `localhost` from `/etc/hosts` it returned just a single IP address.

I have implemented a junky little fix that just compares one `in_addr` to the previous one in `str2paddr` and just skips over any address that is the same as the previous one. There are still cases where this might not detect duplicates, but it should work for 90% of cases where this corner-case problem arises to being with.
